src/Tasks/Microsoft.NET.Build.Tasks/AppHost.cs

```diff
@@ -178,5 +200,51 @@ private static unsafe void Pad0(byte[] searchPattern, byte[] patternToReplace, b
                 }
             }
         }
+
+        /// <summary>
+        /// If the apphost file is a windows PE file (checked by looking at the first few bytes)
+        /// this method will set its subsystem to GUI.
+        /// </summary>
+        /// <param name="accessor">The memory accessor which has the apphost file opened.</param>
+        /// <param name="appHostSourcePath">The path to the source apphost.</param>
+        private static unsafe void SetWindowsGraphicalUserInterfaceBit(
+            MemoryMappedViewAccessor accessor,
+            string appHostSourcePath)
+        {
+            byte* pointer = null;
+
+            try
+            {
+                accessor.SafeMemoryMappedViewHandle.AcquirePointer(ref pointer);
+                byte* bytes = pointer + accessor.PointerOffset;
+
+                // https://en.wikipedia.org/wiki/Portable_Executable
+                // Validate that we're looking at Windows PE file
+                if (((UInt16*)bytes)[0] != 0x5A4D)
```

Comment:
Can you put these numbers into consts? They can be local and close by first use and keeping the comments to documentation nearby as well.